clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -109,9 +109,9 @@ public Metrics(MetricConfig defaultConfig, List<MetricsReporter> reporters, Time
      */
     public Metrics(MetricConfig defaultConfig, List<MetricsReporter> reporters, Time time, boolean enableExpiration) {
         this.config = defaultConfig;
-        this.sensors = new CopyOnWriteMap<>();
-        this.metrics = new CopyOnWriteMap<>();
-        this.childrenSensors = new CopyOnWriteMap<>();
+        this.sensors = new ConcurrentHashMap<>();
```

Comment:
I actually think we should stick to the simplest possible solution i.e. use a ConcurrentHashMap for now. Using a read/write lock would block all reads when adding sensors which is probably not desirable.

For reference, I noticed that yammer metrics uses a ConcurrentHashMap as well for storing all metrics. I think that is a strong point of reference.